starknet_py/serialization/data_serializers/enum_serializer.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+from typing import Dict, Generator, OrderedDict, Tuple, Union
+
+from starknet_py.serialization._context import (
+    DeserializationContext,
+    SerializationContext,
+)
+from starknet_py.serialization.data_serializers.cairo_data_serializer import (
+    CairoDataSerializer,
+)
+from starknet_py.serialization.tuple_dataclass import TupleDataclass
+
+
+@dataclass
+class EnumSerializer(CairoDataSerializer[Union[Dict, TupleDataclass], TupleDataclass]):
+    """
+    Serializer of enums.
+    Can serialize a dictionary and TupleDataclass.
+    Deserializes data to a TupleDataclass.
+
+    Example:
+        {"a": 1} => [0, 1]
+        {"b": 100} => [1, 100]
+        TupleDataclass(variant='a', value=100) => [0, 100]
```

Comment:
Probably adding an example Enum would make sense here. Without it, it's not exactly clear, why `"a"` serializes to 0 and `"b"` serializes to 1.